internal/ethapi/api.go

```diff
@@ -1326,14 +1326,22 @@ func DoEstimateGas(ctx context.Context, b Backend, args TransactionArgs, blockNr
 	// Execute the binary search and hone in on an executable gas limit
 	for lo+1 < hi {
 		mid := (hi + lo) / 2
-		failed, _, err := executable(mid)
+		failed, result, err := executable(mid)
 
 		// If the error is not nil(consensus error), it means the provided message
 		// call or transaction will never be accepted no matter how much gas it is
 		// assigned. Return the error directly, don't struggle any more.
 		if err != nil {
 			return 0, err
 		}
+
+		if result != nil && result.Err != vm.ErrOutOfGas {
```

Comment:
`if result != nil && !errors.Is(result.Err, vm.ErrOutOfGas)` may be better

Reply:
Fixed